scripts/microbe_stage/microbe_editor/microbe_editor.as

```diff
@@ -617,18 +617,22 @@ class MicrobeEditor{
         if(action.cost != 0)
             if(!takeMutationPoints(action.cost))
                 return;
-        // We resize always and insert at the index so that
-        // if we undo something then add something, the new
-        // action is in the right spot on the array.
-        // since we only enqueue when we add new actions
-        actionHistory.resize(actionHistory.length()+1);
+
+        // Every time we make a new action if is in the middle of history of actions
+        // We have to erase all redo history from that point
+        if( actionIndex < actionHistory.length() ) {
```

Comment:
Using a while loop calling actionHistory.removeLast could remove this and be cleaner:
```
while(actionIndex < actionHistory.length())
    actionHistory.removeLast();
```